### PR TITLE
[FIX][ACPRB] fix issue that won't allow multiple payments to be posted…

### PR DESCRIPTION
**Pull request by andres-pcg**
Made it multi records

### CODE REVIEW

**Comment by andrp92** [comment]
Here’s is my signed ICLA,

Regards, 

Sent from Mail for Windows 10

File: account_check_printing_report_base/models/account_payment.py [hunk 1]
```diff
@@ -36,8 +36,10 @@ def do_print_checks(self):
 
     @api.multi
     def post(self):
-        res = super(AccountPayment, self).post()
-        if (self.journal_id.check_print_auto and
-                self.payment_method_id.code == 'check_printing'):
-            return self.do_print_checks()
+        for rec in self:
+            res = super(AccountPayment, rec).post()
```
Comment (pedrobaeza) [inline account_check_printing_report_base/models/account_payment.py]: Don't nest `super` call

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
The CLA should be sent to the email specified in the page.

Check flake8:
```
account_check_printing_report_base/models/account_payment.py:45:1: W391 blank line at end of file
```

File: account_check_printing_report_base/models/account_payment.py (hunk 1 above)
Comment (andres-pcg) [reply account_check_printing_report_base/models/account_payment.py]: Can one of you guys re-factor it? i have already tried several ways to try make the same effect without having to nest it but it didn't work. thanks for the review thou.

Comment (andres-pcg) [reply account_check_printing_report_base/models/account_payment.py]: @pedrobaeza , I have about 3 clients having issues due to this function not being properly implemented to handle multiple payments. 

"Error:
Odoo Server Error

Traceback (most recent call last):
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 650, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/odoo/odoo-server/src/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 692, in dispatch
    result = self._call_function(**self.params)
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/odoo/odoo-server/src/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 936, in _call_
    return self.method(*args, **kw)
  File "/odoo/odoo-server/src/odoo/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/odoo/odoo-server/src/odoo/addons/web/controllers/main.py", line 938, in call_button
    action = self._call_kw(model, method, args, {})
  File "/odoo/odoo-server/src/odoo/addons/web/controllers/main.py", line 926, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/odoo/odoo-server/src/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/odoo/odoo-server/src/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/odoo/odoo-server/src/odoo/addons/account/models/account_payment.py", line 235, in create_payments
    payments.post()
  File "/odoo/odoo-server/src/custom/iterativo/dominicana-premium/invoice_payment_method/models/account_payment.py", line 44, in post
    res = super(AccountPayment, self).post()
  File "/odoo/odoo-server/src/custom/oca/account-payment/account_check_printing_report_base/models/account_payment.py", line 40, in post
    if (self.journal_id.check_print_auto and
  File "/odoo/odoo-server/src/odoo/odoo/fields.py", line 935, in _get_
    record.ensure_one()
  File "/odoo/odoo-server/src/odoo/odoo/models.py", line 4366, in ensure_one
    raise ValueError("Expected singleton: %s" % self)
ValueError: Expected singleton: account.payment(1613, 1614)"

Comment (pedrobaeza) [reply account_check_printing_report_base/models/account_payment.py]: You only have to let the super call as it was, and after that, do the for loop

Comment (andres-pcg) [reply account_check_printing_report_base/models/account_payment.py]: The thing is, that it doesn't even get to the loop if i don't make the super call inside it. It raise an error at the moment of the assignment. when i debugged, here was where the process stopped. 
[   res = super(AccountPayment, self).post() ]

Comment (pedrobaeza) [reply account_check_printing_report_base/models/account_payment.py]: But then the problem is not in this module, but in a super one.

Comment (andres-pcg) [reply account_check_printing_report_base/models/account_payment.py]: I made the changes and tested it and it worked. will update this. thanks .!!

**Comment by andres-pcg** [comment]
@pedrobaeza thanks for your suggestions. Please check the updated version. Thanks.!

File: account_check_printing_report_base/models/account_payment.py [hunk 2]
```diff
@@ -37,7 +37,8 @@ def do_print_checks(self):
     @api.multi
     def post(self):
         res = super(AccountPayment, self).post()
-        if (self.journal_id.check_print_auto and
-                self.payment_method_id.code == 'check_printing'):
-            return self.do_print_checks()
+        for rec in self:
+            if (rec.journal_id.check_print_auto and
+                    rec.payment_method_id.code == 'check_printing'):
+                return rec.do_print_checks()
```
Comment (pedrobaeza) [inline account_check_printing_report_base/models/account_payment.py]: The problem now is that doing a return breaks the loop, not completing the task. First question: `do_print_checks` admit several records? If so just do:
```python
res = super(...)
recs = self.filtered(lambda x: x.journal_id.check_print_auto and
                     x.payment_method_id.code == 'check_printing')
if recs:
    return recs.do_print_checks()
return res
```
If not, simply remove the `return` word.

Comment (andres-pcg) [reply account_check_printing_report_base/models/account_payment.py]: yes, do_print_checks admits multiple records but i found that it also has a return statement inside it, therefor the loop is going to be broken in there as well... 

I think a good solution might be removing the return word..  or maybe this should be checked out deeper. 

what do you think ?

Comment (andres-pcg) [reply account_check_printing_report_base/models/account_payment.py]: Fixed and tested. 

Thanks.

**Comment by andres-pcg** [comment]
Re-factored code following your advice, please check @pedrobaeza.

thanks.

**Comment by andres-pcg** [comment]
@pedrobaeza am i missing anything here?

**Comment by pedrobaeza** [comment]
Travis is still red:
```
account_check_printing_report_base/models/account_payment.py:41:40: E127 continuation line over-indented for visual indent
account_check_printing_report_base/models/account_payment.py:41:80: E501 line too long (84 > 79 characters)
account_check_printing_report_base/models/account_payment.py:45:1: W391 blank line at end of file
account_check_printing_report_base/models/account_payment.py:45:1: W293 blank line contains whitespace
```

**Comment by andres-pcg** [comment]
Thanks. Done. @pedrobaeza

**Review by bodedra — APPROVED** [review]
Code Review :+1:

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**